addons/mail/models/mail_thread.py

```diff
@@ -1313,6 +1313,7 @@ def _message_route_process(self, message, message_dict, routes):
             if thread_id and hasattr(ModelCtx, 'message_update'):
                 thread = ModelCtx.browse(thread_id)
                 thread.message_update(message_dict)
+                message_source = 'email_to_thread'
```

Comment:
All this should be removed and you should override the rights methods: message_new, message_update, ... and you can also use message_type field of mail.message, which is email for incoming email, comment for user input, notification when generated, ...

Reply:
Thank you so much for the suggestion, I'm now using the `message_type` and it seems to work great!